chia/wallet/wallet_transaction_store.py

```diff
@@ -333,18 +333,32 @@ async def get_unconfirmed_for_wallet(self, wallet_id: int) -> List[TransactionRe
         else:
             return []
 
-    async def get_transactions_between(self, wallet_id: int, start, end) -> List[TransactionRecord]:
+    async def get_transactions_between(self, wallet_id: int, start, end, version: int) -> List[TransactionRecord]:
         """Return a list of transaction between start and end index. List is in reverse chronological order.
         start = 0 is most recent transaction
         """
         limit = end - start
-        cursor = await self.db_connection.execute(
-            f"SELECT * from transaction_record where wallet_id=? and confirmed_at_height not in"
-            f" (select confirmed_at_height from transaction_record order by confirmed_at_height"
-            f" ASC LIMIT {start})"
-            f" order by confirmed_at_height DESC LIMIT {limit}",
-            (wallet_id,),
-        )
+
+        # use rowid as additional order by ensures that the returned list is deterministic
+        # otherwise rows with the same height could be returned in a different order
+
+        # Version 1 and Version 2 queries do not return the same data due to sorting and ordering differences
+        if version == 1:
+            cursor = await self.db_connection.execute(
+                f"SELECT * from transaction_record where wallet_id=? and confirmed_at_height not in"
+                f" (select confirmed_at_height from transaction_record order by confirmed_at_height"
+                f" ASC LIMIT {start})"
+                f" order by confirmed_at_height DESC LIMIT {limit}",
+                (wallet_id,),
+            )
+        else:
+            cursor = await self.db_connection.execute(
+                f"SELECT * from transaction_record where wallet_id=? "
+                f"order by confirmed ASC, confirmed_at_height DESC, rowid"
```

Comment:
so, the ordering here is first, all confirmed transactions, and within those, order by height. start with more recent (higher) heights. It's not clear to me why you order by `confirmed` at all.. doesn't a non-null `confirmed_at_height` imply that it's also `confirmed?

Reply:
Yes, I don't think the order by confirmed is required, I was making it implicit that pending will be first in the list. I can test it without as it makes the query simpler.

Reply:
At least in the test code, pending txs have 0 in `confirmed_at_height`, which would put them at the end when sorting DESC, but they would seem to be "newer" than confirmed items, so perhaps they should go first. So if showing newest first, are they newer and therefore should be listed first? This is currently what this PR does.